Assets/MixedRealityToolkit/_Core/Devices/OpenVR/WindowsMixedRealityOpenVRController.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Devices;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Devices.OpenVR
+{
+    /// <summary>
+    /// Open VR Implementation of the Windows Mixed Reality Motion Controllers.
+    /// </summary>
+    public class WindowsMixedRealityOpenVRController : GenericOpenVRController
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="trackingState"></param>
+        /// <param name="controllerHandedness"></param>
+        /// <param name="inputSource"></param>
+        /// <param name="interactions"></param>
+        public WindowsMixedRealityOpenVRController(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+            : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <summary>
+        /// The Generic OpenVR Controller default interactions.
+        /// </summary>
+        /// <remarks>A single interaction mapping works for both left and right controllers.</remarks>
+        public new static readonly MixedRealityInteractionMapping[] DefaultLeftHandedInteractions =
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(1, "Spatial Grip", AxisType.SixDof, DeviceInputType.SpatialGrip, MixedRealityInputAction.None),
+            new MixedRealityInteractionMapping(2, "Trigger Position", AxisType.SingleAxis, DeviceInputType.Trigger, ControllerMappingLibrary.MIXEDREALITY_AXIS9),
+            new MixedRealityInteractionMapping(3, "Trigger Press (Select)", AxisType.Digital, DeviceInputType.TriggerPress,  KeyCode.JoystickButton14),
+            new MixedRealityInteractionMapping(4, "Grip Button", AxisType.Digital, DeviceInputType.ButtonPress, KeyCode.JoystickButton4),
```

Comment:
Nothing is currently coming across many of these axes in Unity 2018. Adding this breaks WMR usability in OpenVR, until Unity fixes their bug.

Reply:
Isn't there an alt keycode to use?

The means generic OVR input would be broken as well, correct?

Reply:
The generic axes are still working, it's just the WMR-specific axes. We could use those here if we want, though touchpad position doesn't seem to be coming through any axis at the moment, since it's the component that's specific to WMR (a thumbstick and a touchpad together).

Reply:
No Bueno